models/OrderDepositTransaction.go

```diff
@@ -0,0 +1,74 @@
+package models
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/delta/dalal-street-server/utils"
+)
+
+type OrderDepositTransaction struct {
+	TransactionId uint32 `gorm:"column:transactionId;not null" json:"transaction_id"`
+	OrderId       uint32 `gorm:"column:orderId;not null" json:"bid_id"`
+	IsAsk         bool   `gorm:"column:isAsk;not null" json:"is_ask"`
+	CreatedAt     string `gorm:"column:createdAt;not null" json:"created_at"`
+}
+
+func (OrderDepositTransaction) TableName() string {
+	return "OrderDepositTransactions"
+}
+
+// GetOrderDepositTransactionRef returns a reference of
+func GetOrderDepositTransactionRef(transactionID, orderID uint32, isAsk bool) *OrderDepositTransaction {
+	return &OrderDepositTransaction{
+		TransactionId: transactionID,
+		OrderId:       orderID,
+		IsAsk:         isAsk,
+		CreatedAt:     utils.GetCurrentTimeISO8601(),
+	}
+}
+
+// GetPlaceOrderTransactionDetails returns price and quantity at which PlaceOrderTransaction was created
+func GetPlaceOrderTransactionDetails(orderID uint32, isAsk bool) (int64, int64, error) {
+	var l = logger.WithFields(logrus.Fields{
+		"method": "GetPlaceOrderTransactionDetails",
+		"userID": orderID,
+	})
+
+	db := getDB()
+	sql := "SELECT transactionId FROM OrderDepositTransactions WHERE orderID = ? and isAsk = ?"
+	rows, err := db.Raw(sql, orderID, isAsk).Rows()
+	if err != nil {
+		l.Errorf("Error retrieving transactionId. Error: %+v", err)
+		return 0, 0, err
+	}
+
+	defer rows.Close()
+	if !rows.Next() {
+		return 0, 0, InvalidOrderIDError{}
+	}
+
+	var transactionId uint32
+	rows.Scan(&transactionId)
+
+	l.Infof("Retrieving reserved asset for transactionId %d", transactionId)
+
+	sql = "SELECT total, stockQuantity from Transactions WHERE id = ?"
```

Comment:
You can write a single query for this no, why are you performing JOIN in Go code when you can use SQL for that?

Reply:
+1

Reply:
Fixed :+1: